src/gui/MainWindow.cpp

```diff
@@ -306,6 +311,11 @@ void MainWindow::changeEvent(QEvent *event)
 
 void MainWindow::captureChanged()
 {
+	if (mCaptureHandler->isSaved() && mCaptureHandler->isPathValid()) {
```

Comment:
- It checks if the capture was saved so it will not add it on every file change.
- It will add it on save / save as.
    - On save, it will add it only if the file is not already in the list of recently opened files

Reply:
I think the SaveOperation would be the correct place for this. Also, while you're there, check if you can write any tests for this, we should try to improve our test coverage.